### PR TITLE
Make autoreload checking more robust

### LINKED ISSUE #6007: No autoreload when saving a file on SAMD21 sometimes
Since around 7.0 I have noticed on SAMD21 boards that sometimes the autoreload doesn't not happen.
It is hard to reproduce and seems to happen randomly. But once the boards stops running autoreload:
- saving files works, but does not trigger autoreload
- ctrl-C ctrl-D works, but does not bring back autoreload
Only a hard reset seems to bring it back.

I have seen the board enter that bad state:
- as soon as I plugged it in.
- after doing some file operations from the host side.
- after leaving the board plugged in without using it for an hour.

I have done some of those tests on a freshly-flashed, latest-main, and erased-file-system CircuitPlayground Express with a simple neopixel animation code.

### PR DESCRIPTION
**Pull request by dhalbert**
- Fixes #6007.

- Add reset for autoreload. De-request ticks.
- Separate state a little more in autoreload.c
- Rename some routines and variables.
- Reset `_tick_event_channel` when ticks are disabled, in atmel_samd/supervisor/port.c
- Add `PYEXEC_RELOAD` as a specific reason for a forced termination of code, instead of checking the exception.
- Remove redundant settings of `CIRCUITPY_AUTORELOAD_DELAY_MS`.

The most vital changes are the autoreload reset, and perhaps the `_tick_event_channel`.

Saving a file over and over again very quickly can still occasionally cause some filesystem corruption, though it seems rare. Interestingly, I found that increasing the autoreload timeout (`CIRCUITPY_AUTORELOAD_DELAY_MS`) from 0.5 to 1 second seems to help a lot for that. It may be that 1 second may be a more realistic waiting time for a write to finish. But it causes a noticeable difference in how fast the board seems to respond, so I did not include it in this PR. But it's something we should consider.

### CODE REVIEW

File: ports/stm/boards/stm32f411ce_blackpill_with_flash/mpconfigboard.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+`/*
```
Comment (tannewt) [inline ports/stm/boards/stm32f411ce_blackpill_with_flash/mpconfigboard.h:1]:
```suggestion
/*
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Everything looks good to me. Just one typo that should break the CI.

**Comment by dhalbert** [comment]
CI failure is GitHub Action's problem.

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**